ground/src/test/java/com/google/android/ground/PermissionDeniedDialogTest.kt

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.android.ground
+
+import androidx.activity.ComponentActivity
+import androidx.compose.ui.test.assertIsDisplayed
+import androidx.compose.ui.test.junit4.createAndroidComposeRule
+import androidx.compose.ui.test.onNodeWithText
+import androidx.compose.ui.test.performClick
+import dagger.hilt.android.testing.HiltAndroidTest
+import kotlin.test.Test
+import org.junit.Rule
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+
+@HiltAndroidTest
+@RunWith(RobolectricTestRunner::class)
+class PermissionDeniedDialogTest : BaseHiltTest() {
+
+  /**
+   * composeTestRule has to be created in the specific test file in order to access the required
+   * activity. [composeTestRule.activity]
+   */
+  @get:Rule override val composeTestRule = createAndroidComposeRule<ComponentActivity>()
+
+  @Test
+  fun permissionDeniedDialog_DisplaysCorrectTitle() {
+    composeTestRule.setContent {
+      PermissionDeniedDialog(signupLink = "http://example.com", onSignOut = {}, onCloseApp = {})
```

Comment:
`onSignOut` and `onCloseApp` are optional parameters and can be skipped. Same comment throughout the class.

Reply:
no, these are not optional